lib/slack/web/api/endpoints/chat.rb

```diff
@@ -236,8 +236,11 @@ def chat_scheduleMessage(options = {})
           # @see https://api.slack.com/methods/chat.unfurl
           # @see https://github.com/slack-ruby/slack-api-ref/blob/master/methods/chat/chat.unfurl.json
           def chat_unfurl(options = {})
-            raise ArgumentError, 'Required arguments :channel missing' if options[:channel].nil?
-            raise ArgumentError, 'Required arguments :ts missing' if options[:ts].nil?
+            if options[:channel].nil? || options[:ts].nil?
+              if options[:unfurl_id].nil? || options[:source].nil?
+                raise ArgumentError, 'Either a combination of :channel and :ts or :unfurl_id and :source is required'
+              end
+            end
```

Comment:
This code is generated, so you'll have to either make a patch in https://github.com/slack-ruby/slack-ruby-client/tree/master/lib/slack/web/api/patches, or figure out a way to specify mutually exclusive options.

Reply:
Note the code as is would allow `channel` and `source`, so you will want to do something else.